homeassistant/components/homematicip.py

```diff
@@ -0,0 +1,103 @@
+"""
+Support for HomematicIP components.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/homematicip/
+"""
+
+import logging
+from datetime import timedelta
+from socket import timeout
+
+import voluptuous as vol
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['homematicip==0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homematicip'
+
+CONF_NAME = 'name'
+CONF_ACCESSPOINT = 'accesspoint'
+CONF_AUTHTOKEN = 'authtoken'
+CONF_TIMEOUT = 'timeout'
+CONF_RECONNECT = 'reconnect'
+CONF_DISCOVERY = 'discovery'
+
+CONFIG_SCHEMA = vol.Schema({
+    vol.Optional(DOMAIN): [vol.Schema({
+        vol.Optional(CONF_NAME, default=''): cv.string,
+        vol.Required(CONF_ACCESSPOINT): cv.string,
+        vol.Required(CONF_AUTHTOKEN): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=timedelta(seconds=20)): (
+            vol.All(cv.time_period, cv.positive_timedelta)),
+        vol.Optional(CONF_RECONNECT, default=timedelta(minutes=10)): (
+            vol.All(cv.time_period, cv.positive_timedelta)),
+        vol.Optional(CONF_DISCOVERY, default=True): cv.boolean,
+    })],
+}, extra=vol.ALLOW_EXTRA)
+
+EVENT_HOME_CHANGED = 'homematicip_home_changed'
+EVENT_DEVICE_CHANGED = 'homematicip_device_changed'
+EVENT_GROUP_CHANGED = 'homematicip_group_changed'
+EVENT_SECURITY_CHANGED = 'homematicip_security_changed'
+EVENT_JOURNAL_CHANGED = 'homematicip_journal_changed'
+
+
+def setup(hass, config):
+    """Set up the HomematicIP component."""
+    # pylint: disable=import-error, no-name-in-module
+    from homematicip.home import Home
+    hass.data.setdefault(DOMAIN, {})
+    homes = hass.data[DOMAIN]
+    accesspoints = config.get(DOMAIN, [])
+
+    @callback
```

Comment:
Callback means it' a function that is okay to run inside the event loop. Homematic doesn't respect this decorator so just remove it.